### PR TITLE
unit tests: Run all tests through Valgrind's memcheck, if available.

### PR DESCRIPTION
**Pull request by tokkee**
Also, fix all problems identified that way.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -57,6 +57,9 @@ AC_PROG_LIBTOOL
 AC_PROG_LEX
 AC_PROG_YACC
 
+AC_PATH_PROG([VALGRIND], [valgrind])
+AC_SUBST([VALGRIND])
```
Comment (rubenk) [inline configure.ac]: Unless I'm mistaken AC_PATH_PROG already calls AC_SUBST on the variable.

**Comment by rubenk** [comment]
One minor nit, otherwise nice!

File: configure.ac (hunk 1 above)
Comment (tokkee) [inline configure.ac]: It does, indeed. Fixed here and in two other cases.

**Comment by tokkee** [comment]
Ftr, we already install Valgrind in the test VMs used by Jenkins.

**Comment by tokkee** [comment]
That test failure is addressed in 9d0f2b090ebdbe151360ba69e05d9c60c0930ac7.

**Comment by rubenk** [comment]
Thanks!

**Pull request merged**